local/owr_gst_local_media_source.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2015, David Chou
+ *     Author: David Chou <david74.chou@gmail.com>
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this
+ * list of conditions and the following disclaimer in the documentation and/or other
+ * materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+ * OF SUCH DAMAGE.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include "owr_gst_local_media_source.h"
+
+GST_DEBUG_CATEGORY_EXTERN(_owrgstlocalmediasource_debug);
+#define GST_CAT_DEFAULT _owrgstlocalmediasource_debug
+
+static void owr_gst_local_media_source_class_init  (OwrGstLocalMediaSourceClass *klass);
+static void owr_gst_local_media_source_init        (OwrGstLocalMediaSource *obj);
+static void owr_gst_local_media_source_dispose     (GObject *object);
+static void owr_gst_local_media_source_set_property(GObject *object, guint property_id, const GValue *value, GParamSpec *pspec);
+static void owr_gst_local_media_source_get_property(GObject *object, guint property_id, GValue *value, GParamSpec *pspec);
+static GstElement *owr_gst_local_media_source_request_source_element(OwrLocalMediaSource *local_media_source);
+
+enum {
+	PROP_0,
+	PROP_SOURCE,
+	N_PROPERTIES
+};
+
+static GParamSpec *obj_properties[N_PROPERTIES] = {NULL, };
+
+struct _OwrGstLocalMediaSourcePrivate {
+	GstElement *source;
+};
+
+#define OWR_GST_LOCAL_MEDIA_SOURCE_GET_PRIVATE(o)      (G_TYPE_INSTANCE_GET_PRIVATE((o), \
+                                                        OWR_TYPE_GST_LOCAL_MEDIA_SOURCE, \
+                                                        OwrGstLocalMediaSourcePrivate))
+
+G_DEFINE_TYPE(OwrGstLocalMediaSource, owr_gst_local_media_source, OWR_TYPE_LOCAL_MEDIA_SOURCE)
+
+static void
+owr_gst_local_media_source_class_init(OwrGstLocalMediaSourceClass *klass)
+{
+	GObjectClass *gobject_class = G_OBJECT_CLASS(klass);
```

Comment:
Did you use tabs? I'm not sure because Github wouldn't show them but please use the same coding style as in the rest of the code base :)